cmd/thanos/compact.go

```diff
@@ -458,6 +460,56 @@ func runCompact(
 		return cleanPartialMarked()
 	}
 
+	if conf.compactionProgressMetrics {
+		g.Add(func() error {
+			unRegisterer := &receive.UnRegisterer{Registerer: reg}
+			ps := compact.NewCompactionProgressCalculator(unRegisterer, tsdbPlanner)
+			var ds *compact.DownsampleProgressCalculator
+			if !conf.disableDownsampling {
+				ds = compact.NewDownsampleProgressCalculator(unRegisterer)
+			}
+
+			return runutil.Repeat(5*time.Minute, ctx.Done(), func() error {
+
+				if err := sy.SyncMetas(ctx); err != nil {
+					return errors.Wrapf(err, "could not sync metas")
+				}
+
+				originalMetas := sy.Metas()
+				groups, err := grouper.Groups(originalMetas)
+				if err != nil {
+					return errors.Wrapf(err, "could not group original metadata")
+				}
+
+				if !conf.disableDownsampling {
```

Comment:
Can you put this code block after compaction progress is done?

Reply:
And we can reduce indentation by:
```go
if conf.disableDownsampling {
  return nil
  }  
```